R/plotStorms.R

```diff
@@ -288,54 +298,142 @@ plotStorms <- function(sts,
-      col = col,
-      lty = lty,
-      pch = pch,
-      lwd = lwd,
-      cex = 0.6,
-      bty = "n"
+  }else {
+    #Init map
+
+    map <- leaflet::fitBounds(
+      leaflet::addProviderTiles(
+        leaflet::leaflet(options =
+                           leaflet::leafletOptions(worldCopyJump = F,
+                                                   minZoom = 2,
+                                                   maxZoom = 12),
+                         width = 650,
+                         height = 700),
+        leaflet::providers$Esri.NatGeoWorldMap,
+        group = "Satellite",
+        options = leaflet::providerTileOptions(errorTileUrl = "Tile not found")),
+      lng1 = as.numeric(ext$xmin),
+      lng2 = as.numeric(ext$xmax),
+      lat1 = as.numeric(ext$ymin),
+      lat2 = as.numeric(ext$ymax)
     )
+
+    #Plotting loi
+    if(loi)
+      map <- leaflet::addPolylines(map,
+                                   data = sts@spatialLoiBuffer,
+                                   fillColor = "transparent",
+                                   label = "Buffer limit")
+
+    if(loi & as.character(sf::st_geometry_type(sts@spatialLoi)) == "POINT")
+      map <- leaflet::addCircleMarkers(map,
+                                       data = sts@spatialLoi,
+                                       fillColor = "transparent",
+                                       label = "LOI")
+
+
+    #Plotting track(s) and labels
+    for (st in stsAux) {
+
+      data <- st@obs.all
+      data$type <- unlist(lapply(data$msw,getColors, sshs)) # TODO ???
+
+      labels <- paste0("<b>",st@name, "</b><br>",
+                       "<i>observation ",row.names(data), "</i><br>",
+                       data$iso.time,
+                       "<ul>",
+                       "<li>", "scale:  ", data$sshs, "</li>",
+                       "<li>", "msw:  ", data$msw, "m/s</li>",
+                       "<li>", "rmw:  ", data$rmw, "km</li>",
+                       "<li>", "pressure:  ", data$pressure, "pa</li>",
+                       "<li>", "poci:  ", data$poci, "pa</li>",
+                       "</ul>")
+
+      #Plot track
+      map <- leaflet::addPolylines(map,
+                                   data = data,
+                                   lng = ~lon,
+                                   lat = ~lat,
+                                   color = "black",
+                                   weight = 2)
+
+
+      map <- leaflet::addCircleMarkers(map,
+        data = data,
+        lng = ~lon,
+        lat = ~lat,
+        radius = 5,
+        color = ~type,
+        stroke = FALSE,
+        fillOpacity = 1,
+        popup = labels
+      )
+    }
+
+    #Adding legends
+    map <- leaflet::addLegend(map,
+                              legends,
+                              colors = sshsPalette,
+                              labels = names(sshsPalette),
+                              title = "SSHS",
+                              opacity = 1)
+
+    map
```

Comment:
Why do you need this line ?

Reply:
I don't remember for the moment, I think it is actually what creates the plot